esp/services/ws_fileio/ws_fileioservice.cpp

```diff
@@ -49,14 +49,17 @@ bool CWsFileIOEx::CheckServerAccess(const char* targetDZNameOrAddress, const cha
         Owned<IConstDropZoneInfoIterator> dropZoneItr = env->getDropZoneIteratorByAddress(targetDZNameOrAddress);
         ForEach(*dropZoneItr)
         {
-            dropZoneInfo.set(&dropZoneItr->query());
-            break;
+            IConstDropZoneInfo & dz = dropZoneItr->query();
+            if (dz.isECLWatchVisible())
+            {
+                dropZoneInfo.set(&dropZoneItr->query());
+                break;
+            }
         }
     }
 
     if (dropZoneInfo)
     {
-        char ch = '\\';
         SCMStringBuffer dropZoneName, directory, computerName, computerAddress;
         dropZoneInfo->getName(dropZoneName);
         dropZoneInfo->getComputerName(computerName); //legacy structure
```

Comment:
@rpastrana If you find the dropzone using getDropZoneIteratorByAddress(), you already know the network address. Then, you should be able to skip line 65 to 88.